src/middleware/ipFilter.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+module.exports = setup;
+
+const _ = require('lodash');
+const iputil = require('ip');
+const rangeCheck = require('range_check');
+const createError = require('http-errors');
+
+ /**
+  * testExplicitIp Test an Explicit IP against the list of constraints
+  * @param  {String} ip          IP address of caller
+  * @param  {String} constraints IPs to restrict or allow
+  * @param  {String} mode        mode of operation deny/allow
+  * @return {Boolean}             True or False, if we should block or allow and matched
+  */
+function testExplicitIp(ip, constraints, mode) {
```

Comment:
I would also call the second variable more specifically in each of the 3 function, here it's a targetIP, in the second its a CIDRBlock, in the third its a range